File: app/src/main/java/com/yanftch/review/android/fragment/new_home/BaseView.java
package com.yanftch.review.android.fragment.new_home;

import android.content.Context;

public interface BaseView<T> {

    // 注入presenter
    void setPresenter(T presenter);

    // 判断当前view是否存活，异步的时候一定要判断
    boolean isActive();

    Context getViewContext();
}

File: app/src/main/java/com/yanftch/review/android/base/BaseView.kt
package com.yanftch.review.android.base

/**
 *
 * User : yanftch
 * Date : 2020-01-29
 * Time : 20:57
 * Desc : base view
 */
interface BaseView<T> {
    fun setPresenter(presenter: T)
    fun isActive(): Boolean
}